### PR TITLE
ci: enable Win ARM

### PR DESCRIPTION
**Pull request by henryiii**
Will un-draft when I download the produced wheel and test locally.

### CODE REVIEW

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -48,6 +48,9 @@ jobs:
         # Cross-compile on macOS
         CIBW_ARCHS_MACOS: x86_64 arm64
 
+        # Cross-compile on Win
+        CIBW_ARCHS_WIN: AMD64 ARM64
```
Comment (henryiii) [inline .github/workflows/wheels.yml]:
```suggestion
        CIBW_ARCHS_WINDOWS: AMD64 ARM64
```

File: .github/workflows/wheels.yml [hunk 2]
```diff
@@ -45,11 +48,20 @@ jobs:
 
     - uses: pypa/cibuildwheel@v2.13.0
       env:
+        CIBW_BUILD: cp311-* cp312-*
+      
         # Cross-compile on macOS
         CIBW_ARCHS_MACOS: x86_64 arm64
 
+        # Cross-compile on Win
+        CIBW_ARCHS_WINDOWS: AMD64 ARM64
```
Comment (henryiii) [inline .github/workflows/wheels.yml]:
```suggestion
        CIBW_ARCHS_WINDOWS: ARM64
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -27,12 +27,14 @@ endif()
...
 # Try to import all Python components potentially needed by nanobind
 find_package(Python 3.8
-  REQUIRED COMPONENTS Interpreter Development.Module
-  OPTIONAL_COMPONENTS Development.SABIModule)
+  REQUIRED COMPONENTS Interpreter Development.Module ${SKBUILD_SABI_COMPONENT})
 
 # Import nanobind through CMake's find_package mechanism
 find_package(nanobind CONFIG REQUIRED)
 
+include(CMakePrintHelperes)
```
Comment (henryiii) [inline CMakeLists.txt]:
```suggestion
include(CMakePrintHelpers)
```

File: pyproject.toml [hunk 4]
```diff
@@ -1,5 +1,7 @@
 [build-system]
-requires = ["scikit-build-core >=0.4.3", "nanobind >=1.3.2"]
+requires = [
+    "scikit-build-core @ git+https://github.com/scikit-build/scikit-build-core@henryiii/fix/crosscompile",
+    "nanobind @ git+https://github.com/henryiii/nanobind@henryiii/fix/suffix"]
```
Comment (henryiii) [inline pyproject.toml]:
```suggestion
    "nanobind @ git+https://github.com/wjakob/nanobind@master"]
```

**Comment by wjakob** [comment]
What's the status on this PR? Are we waiting for a future `cibuildwheel` release?

**Comment by henryiii** [comment]
The problem is https://gitlab.kitware.com/cmake/cmake/-/issues/25001. Since nanobind sets the extension separately (like pybind11), and since scikit-build-core does make the correct extension available, we could fix it in nanobind but not generally; maybe that's the best way to go for now.

**Comment by wjakob** [comment]
Checking in on the status of this PR as the CMake blocker issue sounds like it's unlikely to be fixed.

**Comment by henryiii** [comment]
Yeah, there's two options. One would be to check and see if `SKBUILD_SOABI` is defined, then use that. That's always correct, and nanobind could just look for that and override if present. The other would be for scikit-build-core to cause Python to report the cross-compiled values so that FindPython picks it up, and that was hard and the PR is on hold. There continue to be good reasons to allow us to set the SOABI from scikit-build-core, so I'm still holding out hope that there might be an option someday.

But the best course of action for nanobind specifically is to check for SKBUILD_SOABI and use that if set.

**Comment by wjakob** [comment]
Like this? https://github.com/wjakob/nanobind/pull/483. I am not quite following the subtleties as to where `SKBUILD_SO[S]ABI` and/or `Python_SO[S]ABI` is defined. My assumption would have been that they are equivalent when run via scikit-build-core.

**Comment by henryiii** [comment]
That looks correct. They are usually equivalent when running in scikit-build-core, but if they differ, scikit-build-core is pulling the value from the running Python using the standard mechanisms (including respecting cross-compiling) while FindPython is setting Python_SOABI from a subprocess call and it's getting it incorrectly when cross-compiling.

https://github.com/scikit-build/scikit-build-core/pull/355 would fix FindPython by getting the Python subprocess it calls to report the cross-compiling values, but it's tricky to get right.

**Comment by wjakob** [comment]
closing in favor of #43.

**Pull request closed without merging**